### PR TITLE
historical files parsing and +95 degree F days + precip

### PR DESCRIPTION
**Pull request by emileten**
[Note : I edited this PR to contain also another branch's changes]
[Note 2 : added precip changes] 
@dgergel This contains three additions : automating the parsing of historical files, days with F temp > 95 and precip diagnostics. 

Regarding the former addition, note that, for now the code only works with an ssp+historical workflow. If the need arises, it's easy to add in flexibility (by splitting the `dict` in `collect_paths` and create one or the other based on a `bool` argument).

### CODE REVIEW

**Comment by emileten** [comment]
@dgergel this is ready. All the options aren't there for precip, but it's very quick to expand once there is data to look at.

**Review by dgergel — CHANGES_REQUESTED** [review]
This all looks good to me - thanks @emileten! Just a couple changes. Nice work!

File: notebooks/downscaling_pipeline/science_validation_manual.py [hunk 1]
```diff
@@ -197,34 +235,37 @@ def collect_paths(manifest, gcm='GFDL-ESM4', ssp='ssp370', var='tasmax'):
     dict
     """
 
+    non_historical_token = '(?=.*,target:ssp,)'
```
[the remark this answers is not in the document]
Comment (emileten) [reply notebooks/downscaling_pipeline/science_validation_manual.py]: done 👍

File: notebooks/downscaling_pipeline/science_validation_manual.py [hunk 2]
```diff
@@ -197,34 +235,37 @@ def collect_paths(manifest, gcm='GFDL-ESM4', ssp='ssp370', var='tasmax'):
...
-                'historical': 'scratch/biascorrectdownscale-bk6n8/biascorrectdownscale-bk6n8-858077599/out.zarr'
+                ssp: f(manifest, f'{non_historical_token}{var_token}{ssp_token}{gcm_token}(?=.*biascorrect)(?=.*preprocess-simulation)')['path'],
+                'historical': f(manifest, f'{historical_token}{var_token}{ssp_token}{gcm_token}(?=.*biascorrect)(?=.*preprocess-simulation)')['path']#'scratch/biascorrectdownscale-bk6n8/biascorrectdownscale-bk6n8-858077599/out.zarr'
             },
             'bias_corrected': {
-                ssp: f(manifest, f'{var_token}{ssp_token}{gcm_token}(?=.*rechunk-biascorrected)')['path'],
-                'historical': 'az://biascorrected-stage/CMIP/NOAA-GFDL/GFDL-ESM4/historical/r1i1p1f1/day/tasmax/gr1/v20210920214427.zarr'
+                ssp: f(manifest, f'{non_historical_token}{var_token}{ssp_token}{gcm_token}(?=.*rechunk-biascorrected)')['path'],
+                'historical': f(manifest, f'{historical_token}{var_token}{ssp_token}{gcm_token}(?=.*rechunk-biascorrected)')['path']#'az://biascorrected-stage/CMIP/NOAA-GFDL/GFDL-ESM4/historical/r1i1p1f1/day/tasmax/gr1/v20210920214427.zarr'
```
[the remark this answers is not in the document]
Comment (emileten) [reply notebooks/downscaling_pipeline/science_validation_manual.py]: done 👍

**Pull request merged**